File: py/storage/Volume.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import json
from .Node import Node

class Volume(object):
	def __init__(self, path: str):
		self.path = path
		self.root = None
		
		self.__open()
		
	def __open(self):
		
		with open(self.path, "r", encoding="utf-8") as fd:
			jdata = json.load(fd)
			self.__from_dict(jdata)
			
			
	def __from_dict(self, data: dict):
		root_nodes = data.get("root", None)
		
		if root_nodes:
			self.root = Node()
			self.root.parse(root_nodes, re=True)

File: src/storage/Volume.cpp
#include <QFile>
#include <QJsonObject>
#include <QDebug>
#include "Volume.hpp"

Volume::Volume(const QString &ipath){
	this->path = ipath;
}

bool Volume::open(){

	QFile file(this->path);
	if (!file.open(QIODevice::ReadOnly | QIODevice::Text)){
		qDebug() << "vol - unable open file to read: " << this->path;
		return false;
	}


	QByteArray saveData = file.readAll();


	// TODO: parse error
	QJsonDocument doc(QJsonDocument::fromJson(saveData));


	this->__from_json(doc);

	return true;


	return true;
}

void Volume::__from_json(const QJsonDocument &doc){

	QJsonObject doc_obj = doc.object();

	if(doc_obj.contains("root")){
//		this->version = root.value("version").toInt();

		this->root = Node();

	}


	this->root.from_json(doc_obj.value("root").toObject(), true);


//	if(doc_obj.contains("description")){
//		this->description = root.value("description").toString();
//	}
}


